client/ayon_blender/plugins/publish/collect_workfile.py

```diff
@@ -16,7 +16,7 @@ def process(self, instance):
         """Process collector."""
 
         context = instance.context
-        filepath = Path(context.data["currentFile"])
+        filepath = Path(context.data.get("currentFile") or "")
         ext = filepath.suffix
 
         instance.data.update(
```

Comment:
if the filepath is empty string, `ext =filepath.suffix` most likely errors out.

Reply:
I did a test run and it seemed fine. ✅ 

It would maybe be better if current file is none that it'd not end up doing the rest and return - but I'm worried that other plug-ins may be dependent on those attributes and hence we'll run into other errors.

Reply:
Shouldn't be the instance deactivated if filepath is not available?

Reply:
BTW the paht is filled in `"setMembers"` but I couldn't find usage of the key anywhere.

Reply:
This is actually used automatically due to the deadline logic using the `AbstractSubmitDeadline`?

https://github.com/ynput/ayon-deadline/blob/c386aeb2e83f8e484c19d373d57e6ff561ff14b7/client/ayon_deadline/plugins/publish/blender/submit_blender_deadline.py#L29-L31

Reply:
> Shouldn't be the instance deactivated if filepath is not available?

Valid point - we could just set the instance data active to False. That's quite clever actually.

Reply:
It was just first idea, but I don't know hat are consequences? If this add workfile to render instances then it might not be good idea, and maybe let it crash is better actually.

Reply:
> If this add workfile to render instances then it might not be good idea, and maybe let it crash is better actually.

A validator still disallows the actual publish - this is just to avoid errors on Collecting. Made the changes, seemed fine for me.